mysys/my_largepage.c

```diff
@@ -79,88 +70,361 @@ void my_large_free(uchar* ptr)
     my_large_malloc_int(), i.e. my_malloc_lock() was used so we should free it
     with my_free_lock()
   */
-  if (!my_use_large_pages || !my_large_page_size || !my_large_free_int(ptr))
+  if (!my_large_free_int(ptr, size))
     my_free_lock(ptr);
```

Comment:
BTW, what memory we're unlocking here? After successful unmapping it doesn't exist in application. I think it's at least redundant.

Reply:
Sorry, I was confused by `my_large_free_int()` return value. We do unlocking only on `munmap()` failure. At least it's correct.